sklearn/manifold/mds.py

```diff
@@ -271,6 +274,43 @@ def smacof(similarities, metric=True, n_components=2, init=None, n_init=8,
         return best_pos, best_stress
 
 
+def svd_mds(similarities, n_components=2):
+    """
+    Computes multidimensional scaling using SVD algorithm
+
+    Parameters
+    ----------
+    similarities : symmetric ndarray, shape (n_samples, n_samples)
+        similarities between the points
+
+    n_components : int, optional, default: 2
+        number of dimension in which to immerse the similarities
+        overridden if initial array is provided.
+    """
+
+    similarities = check_array(similarities)
+    n_samples = similarities.shape[0]
+
+    if similarities.shape[0] != similarities.shape[1]:
+        raise ValueError("similarities must be a square array (shape=%d)" %
+                         n_samples)
+    if not np.allclose(similarities, similarities.T):
+        raise ValueError("similarities must be symmetric")
+
+    H = np.eye(*similarities.shape) - 1./n_samples*np.ones(similarities.shape)
+    K = -0.5*np.dot(H, np.dot(similarities**2, H))
```

Comment:
Ah, this is centering, right? Maybe using the ``KernelCenterer``? Otherwise doing the column-wise sums and subtracting looks more efficient than doing the matrix products. If you bench, can you maybe compare to using ``KernelCenterer``?